doc/source/ray-core/user-spawn-processes.rst

```diff
@@ -0,0 +1,106 @@
+Lifetimes of a User-Spawn Process
+=================================
+
+To avoid leaking user-spawned processes, Ray provides mechanisms to kill all user-spawned processes when a worker that starts it exits. This feature prevents GPU memory leaks from child processes (e.g., torch).
+
+We have 2 environment variables to handle subprocess killing on core worker exit:
+
+- ``RAY_kill_child_processes_on_worker_exit`` (default ``true``): Only works on Linux. If true, core worker kills all subprocesses on exit. This won't work if the core worker crashed or was killed by a signal. If a process is created as a daemon, e.g. double forked, it will not be killed by this mechanism.
+
+- ``RAY_kill_child_processes_on_worker_exit_with_raylet_subreaper`` (default ``false``): Only works on Linux greater than or equal to 3.4. If true, Raylet kills any subprocesses that were spawned by the core worker after the core worker exits. This works even if the core worker crashed or was killed by a signal. Even if a process is created as a daemon, e.g. double forked, it will still be killed by this mechanism. The killing happens within 10 seconds after the core worker death.
```

Comment:
What's the plan to turn this on by default and remove the other one?

Reply:
I mean removing the other `RAY_kill_child_processes_on_worker_exit` flag and the related code since we don't need two cleanup mechanisms.